Sources/TuistCore/Utils/RootDirectoryLocator.swift

```diff
@@ -74,6 +74,12 @@ public final class RootDirectoryLocator: RootDirectoryLocating {
         } else if try await fileSystem.exists(path.appending(component: ".git"), isDirectory: true) {
             cache(rootDirectory: path, for: source)
             return path
+        } else if try await fileSystem.exists(
+            path.appending(component: Constants.SwiftPackageManager.packageSwiftName),
+            isDirectory: false
+        ) {
+            cache(rootDirectory: path, for: source)
+            return path
```

Comment:
We should not check for `Package.swift` when traversing up. We should only check whether a `Package.swift` exists if we fail to find a root directory.

That is [this](https://github.com/tuist/tuist/pull/7156/files#diff-464d3093b201aa910bf709a26481dc12a26b2b6e12582a6c7f22666008ba75b1R57) method should look something like this:
```swift
    public func locate(from path: AbsolutePath) async throws -> AbsolutePath? {
        if let path = try await locate(from: path, source: path) {
         return path
      } else if try await fileSystem.exists(path.appending(component: Constants.SwiftPackageManager.packageSwiftName), isDirectory: false) {
     return path
    } else {
      return nil
    }
    }
```

Reply:
Test `test_prints_the_manifest_when_workspace_manifest` fails with these changes

crashes with an error
```
Unknown: Unable to get information about '/var/folders/sz/k3wbm63x3tvc21qgm8zvhddj7y8gjf/T/TemporaryDirectory.0InATV/Workspace.swift/Package.swift
```

I'll see if I can figure out why this happened.